### PR TITLE
Added weather forecast on the more info screen

### PR DESCRIPTION
**Pull request by ecornamu**
- Added the possiblity to query `OpenWeatherMap API` for tomorrow's weather prediction on an `Activity`'s location

- Implemented it in the `MoreInfoScreen` : you can now click on the weather to get tomorrow's forecast

image

- Since I wrote code on the `WeatherViewModel` which we deemed untestable, I have a line coverage of 76.6% because of those untestable new lines

### CODE REVIEW

File: app/src/main/java/com/lastaoutdoor/lasta/data/api/weather/WeatherApiResponse.kt
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/data/api/weather/WeatherApiResponse.kt:1]: Great code, and good job using data classes!

File: app/src/main/java/com/lastaoutdoor/lasta/ui/components/WeatherReport.kt [hunk 1]
```diff
@@ -23,10 +25,11 @@ import androidx.compose.ui.text.font.FontWeight
...
 import androidx.compose.ui.unit.sp
 import com.lastaoutdoor.lasta.R
+import com.lastaoutdoor.lasta.data.api.weather.WeatherForecast
 import com.lastaoutdoor.lasta.data.api.weather.WeatherResponse
 import com.lastaoutdoor.lasta.data.api.weather.getWeatherIconFromId
+import com.lastaoutdoor.lasta.data.api.weather.kelvinToCelsius
 import com.lastaoutdoor.lasta.ui.theme.PrimaryBlue
-import java.math.RoundingMode
 
 private const val KELVIN_CONST = 273.15
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/components/WeatherReport.kt:34]: This constant is not used anymore, you could move it in `WeatherApiResponse.kt`

File: app/src/main/java/com/lastaoutdoor/lasta/ui/components/WeatherReport.kt
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/components/WeatherReport.kt:1]: This composable seems well implemented, thank you for putting the report into a Card, it is much more aestethic.

File: app/src/main/java/com/lastaoutdoor/lasta/viewmodel/WeatherViewModel.kt [hunk 2]
```diff
@@ -63,11 +68,33 @@ constructor(
 
       // Call surrounded by try-catch block to make handle exceptions caused by Weather API
       try {
+        // current weather for activity location
         val weather = weatherRepository.getWeatherWithLoc(a.startPosition.lat, a.startPosition.lon)
         _weather.postValue(weather)
+        // forecast for activity location
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/viewmodel/WeatherViewModel.kt:74]: Great reflex to put it inside the try block, so the ErrorToast handle the error :)

**Review by Thimphou — CHANGES_REQUESTED** [review]
Hey Emile!

Your feature will greatly help our users advise their future activities, it is very desirable and it add a lot of value.

Although, I think it is possible to get those 4 remaining percents to achieve 80% line coverage, I'm sure you will successfully do it :)

**Review by CassioManuguerra — CHANGES_REQUESTED** [review]
Solid PR my g !
Having the weather forecasts is such a useful feature to our app, nice job on getting it done ! Your code is really nice, you commented (feels so good), you tested (as much as possible).

It is just a bit sad that you didn't get the 80%, are you sure you cannot test the weatherViewModel ? Please try testing it using a blank Application() call !

Another thing that I think is important would be to implement a back arrow for the small dialog opening, the one on your photo looks a bit "unprofessional" or "drafty".

Thanks in advance for trying to fix these, good job once again,
Cassio

**Comment by ecornamu** [comment]
Hey @Thimphou and @CassioManuguerra, thank you both for your feedback ! I added further testing to increase line coverage to 84% thanks to some big help from @CassioManuguerra. 
The only requested change I didn't implement is the arrow on the Forecast Card, because I don't agree with its addition to the UI and I think we need to discuss this with the team and maybe change this later.

**Review by Thimphou — APPROVED** [review]
Changes are looking good, thank you for your quick  adjustments.

**Review by CassioManuguerra — APPROVED** [review]
Thanks for changing, great job on testing, approving now !!!

**Pull request merged**